pilot/pkg/ambient/controller/waypoint_controller.go

```diff
@@ -130,103 +143,85 @@ func (rc *WaypointProxyController) Reconcile(name types.NamespacedName) error {
 			log.Errorf("unable to fetch Gateway: %v", err)
 			return err
 		}
-		log.Debugf("gateway deleted")
-		return rc.pruneGateway(name)
+		return nil
 	}
 
 	if gw.Spec.GatewayClassName != "istio-mesh" {
-		log.Debugf("mismatched class %q", gw.Spec.GatewayClassName)
-		return rc.pruneGateway(name)
+		istiolog.Debugf("mismatched class %q", gw.Spec.GatewayClassName)
```

Comment:
```suggestion
		log.Debugf("mismatched class %q", gw.Spec.GatewayClassName)
```

Reply:
Also don't we need to remove the Deployment then, if it exists?

Reply:
changing it is supported afaik, and that is the case I was thinking of

Reply:
In this case, how can we ensure that the deployment we delete belongs to us?  If we delete whatever deployment is associated with the gateway, we could well delete other gateway controllers' deployments...

Reply:
Hmm, good point. I suppose we could have some sort of label that declares it but that feels a bit hacky. Maybe we defer this